### ISSUE TITLE
How to view table created by SHC from Phoenix?

### ISSUE THREAD

**Issue by charles138** [hortonworks-spark/shc#168]
In your article “[SPARK HBASE CONNECTOR – A YEAR IN REVIEW](https://hortonworks.com/blog/spark-hbase-connector-a-year-in-review/)”, you mentioned the following:

As Phoenix is the default coder, the only change for the users is that if they want to use PrimitiveType as the data coder, they need to specify “tableCoder”:”PrimitiveType” in their catalogs to notify SHC that they want to use PrimitiveType instead of Phoenix as “tableCoder”.

Does it mean if I remove "tableCoder":"PrimitiveType" in cat and run below, I can view the table “shcExampleTable” in Phoenix? I tried with and without and can’t see the table. Please advise.

`
--spark-shell
export SPARK_HOME=/usr/hdp/current/spark2-client   export SPARK_MAJOR_VERSION=2   cd /usr/hdp/current/spark2-client       spark-shell --master local[*] --packages   com.hortonworks:shc-core:1.1.1-2.1-s_2.11 --num-executors 4 --driver-memory   512m --executor-memory 512m --executor-cores 1 --repositories   http://nexus-private.hortonworks.com/nexus/content/repositories/IN-QA/

--Create Table and Insert Records
val sc = spark.sparkContext
val sqlContext = spark.sqlContext

import sqlContext.implicits._
import spark.sql

val cat = s"""{
|"table":{"namespace":"default", "name":"shcExampleTable", "tableCoder":"PrimitiveType"},
|"rowkey":"key",
|"columns":{
|"key":{"cf":"rowkey", "col":"key", "type":"int"},
|"col1":{"cf":"cf1", "col":"col1", "type":"boolean"},
|"col2":{"cf":"cf2", "col":"col2", "type":"double"},
|"col3":{"cf":"cf3", "col":"col3", "type":"float"}
|}
|}""".stripMargin
     
    
import org.apache.spark.sql.{DataFrame, SparkSession}
import org.apache.spark.sql.execution.datasources.hbase

def withCatalog(cat: String): DataFrame = {
sqlContext
.read
.options(Map(org.apache.spark.sql.execution.datasources.hbase.HBaseTableCatalog.tableCatalog->cat))
.format("org.apache.spark.sql.execution.datasources.hbase")
.load()
}

case class LRRecord(
key: Int,
col1: Boolean,
col2: Double,
col3: Float)

val data = Seq(LRRecord(1.toInt,false, 1.0.toDouble, 1.0.toFloat), LRRecord(2.toInt,true, 1.0.toDouble, 1.0.toFloat))

sc.parallelize(data).toDF.write.options(Map(org.apache.spark.sql.execution.datasources.hbase.HBaseTableCatalog.tableCatalog -> cat, org.apache.spark.sql.execution.datasources.hbase.HBaseTableCatalog.newTable -> "5")).format("org.apache.spark.sql.execution.datasources.hbase").save()

val df1 = withCatalog(cat)

df1.show

--Phoenix
cd /usr/hdp/current/phoenix-client/bin
./sqlline.py sandbox.hortonworks.com

!tables

`

**Comment by weiqingy** [comment]
Hi @charles138 ,

For creating HBase tables, there are some differences between SHC and Phoenix. When Phoenix creates a HBase table, it will create a `"data table"` (stored the 'real' data) and a `"metadata/schema table"` of the `"data table"`. When Phoenix read data from the Hbase table, it will check the `"metadata/schema table"` first, then based on the `"metadata/schema table"`, it will find the `"data table"` and read data from it. However, if you use SHC to create a Hbase table, it will only create `"data table"` in Hbase without the `"metadata/schema table"`. So if you want Phoenix to 'see' SHC tables, you can use the following command line to create the `"metadata/schema table"` first:
```
create table IF NOT EXISTS SHCTABLE(col00 VARCHAR not null, col01 INTEGER not null, cf1.col1 BOOLEAN, cf1.col2 DOUBLE, cf2.col3 FLOAT, cf2.col4 INTEGER, cf3.col5 BIGINT, cf3.col6 SMALLINT, cf4.col7 VARCHAR, cf4.col8 TINYINT CONSTRAINT pk PRIMARY KEY (col00, col01));
```
The SHCTABLE is corresponding to the catalog in [this Phoenix example](https://github.com/weiqingy/shc/blob/mySHC/examples/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/Phoenix1.scala#L55) which I used to test the feature of Phoenix coder.

Thanks.

**Comment by charles138** [comment]
Yes, I created the table in Phoenix, run the codes, and select the table in Phoenix without any problems. Thank you for your prompt reply. One observation: in the 3 Phoenix demo scripts, there are data types for Date and TimeStamp. I assume SHC supports them just not part of the demo. Is this correct?

**Comment by thewilliamzhang** [comment]
I created a table by saving a Spark DataFrame through shc catalog on HBase. Then with sqlline.py, I created a table in phoenix like the example demonstrated above. The create-table statement executed fine and I am able to see the newly created phoenix table through `!table` command. However the new table is empty (has 0 rows by select *). Phoenix could not figure out how to pick up data from HBase. And if I drop this phoenix table the original HBase table is not affected at all.

What could be the issue here?
